pkg/controller/workflow/result.go

```diff
@@ -11,21 +11,34 @@ const defaultRetry = time.Second * 10
 type ConditionReason string
 
 type Result struct {
-	done         bool
+	terminated   bool
 	requeueAfter time.Duration
 	message      string
 	reason       ConditionReason
 }
 
 func OK() Result {
 	return Result{
-		done:         false,
+		terminated:   false,
 		requeueAfter: -1,
 	}
 }
 
 func Terminate(reason ConditionReason, message string) Result {
-	return Result{done: true, requeueAfter: defaultRetry, reason: reason, message: message}
+	return Result{
+		terminated:   true,
+		requeueAfter: defaultRetry,
+		reason:       reason,
+		message:      message,
+	}
+}
+
+func InProgress(message string) Result {
+	return Result{
+		terminated:   false,
```

Comment:
`true`

Reply:
`true` would mean this is not `IsOK()` - are you sure this is right?

Reply:
the main answer is "should the reconciliation end right now or it should continue"
In this case I guess there is no point in continuing the reconciliation as the cluster is not ready yet and we need to retry soon?